test/integration/targets/parsing/runme.sh

```diff
@@ -4,3 +4,9 @@ set -eux
 
 ansible-playbook parsing.yml -i ../../inventory "$@" -e "output_dir=${OUTPUT_DIR}"
 ansible-playbook good_parsing.yml -i ../../inventory "$@"
+
+# test that we don't inject _raw_params with invalid extra vars
+[ "$(ansible-playbook no_extras.yml -i ../../inventory -e 'bad var as string' 2>&1 |grep -c 'Invalid k=v detected')" -gt "0" ]
+[ "$(ansible-playbook no_extras.yml -i ../../inventory -e 'valid=var inkvalidstring' 2>&1 |grep -c 'Invalid k=v detected')" -gt "0" ]
```

Comment:
Update the regex to include the error message generated when catching the `ValueError`. This will ensure the exception handler is working as intended. Also, since the playbook here never runs, just use an ad-hoc task:

```suggestion
[ "$(ansible -m debug testhost -i ../../inventory -e 'bad var as string' 2>&1 |grep -c 'Invalid extra vars option passed. Invalid k=v detected')" -gt "0" ]
[ "$(ansible -m debug testhost -i ../../inventory -e 'valid=var inkvalidstring' 2>&1 |grep -c 'Invalid extra vars option passed. Invalid k=v detected')" -gt "0" ]
```